enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,677 @@
+  `platform.aws.subnets`, even if compute pool was not defined on `compute[?name=="edge"]`
+
+> Note: Local Zones subnet scope. See section "Public subnet and outbound traffic"
+- only `public` subnets are allowed for Local Zone subnets ID added on `platform.aws.subnets`
+- the Local Zone type should not be allowed on the `worker` compute pool
+  `compute[?name=="worker"].platform.aws.zones`.
+
+- the default or user-provided (`compute[?name=="edge"].platform.aws.type` or
+  `platform.aws.defaultMachinePlatform.type`) instance type for the machine
+  pool `compute[?name=="edge"]` should be validated on the EC2 offering for the
+  target availability zone
+
+> Note: Currently the resource types are limited. The intersection between zones is c5d.2xlarge and gp2
+- the default `edge` machine pool instance type must be common in the current
+  availability zone offerings. The default for the `edge` pool will be `c5d.2xlarge`.
+- the default `edge` machine pool EBS type must be common in the current
+  availability zone offerings. The default for the `edge` pool will be `gp2`.
+
+#### Examples configuration for user-provided network
+
+Those are two examples of a user-provided network when installing a cluster
+in existing VPC:
+
+1) `install-config.yaml` with Local Zone subnets, and empty `edge` machine pool
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: edge-cluster
+platform:
+  aws:
+    region: us-east-1
+    subnets:
+    - subnetID-us-east-1a
+    - subnetID-us-east-1b
+    - subnetID-us-east-1c
+    - subnetID-us-east-1-nyc-1a
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+2) `install-config.yaml` with Local Zone subnets, and custom instance type
+
+> Note: the instance type `c5d.2xlarge` common between locations
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: edge-cluster
+compute:
+- name: edge
+  platform:
+    aws:
+      type: c5d.2xlarge
+platform:
+  aws:
+    region: us-east-1
+    subnets:
+    - subnetID-us-east-1a
+    - subnetID-us-east-1b
+    - subnetID-us-east-1c
+    - subnetID-us-east-1-nyc-1a
+    - subnetID-us-east-1-mia-1a
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+3) `install-config.yaml` with Local Zone subnets, and custom volume type
+
+> Note: currently only `us-west-2` supports `gp3`
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: edge-cluster
+compute:
+- name: edge
+  platform:
+    aws:
+      rootVolume:
+        type: gp3
+platform:
+  aws:
+    region: us-west-2
+    subnets:
+    - subnetID-us-west-2a
+    - subnetID-us-west-2b
+    - subnetID-us-west-2c
+    - subnetID-us-west-2-lax-1a
+    - subnetID-us-west-2-lax-1b
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+The `install-config.yaml` exemplified above will generate oen MachineSet
+by subnet ID on install directory `<install-dir>/openshift/`:
+
+Example  `openshift/99_openshift-cluster-api_worker-machineset-3.yaml`:
```

Comment:
Would it make sense to have the compute pool name "edge" in the file name here? E.g.

```
openshift/99_openshift-cluster-api_edge-machineset-3.yaml`
```

Reply:
We could, but it's only for the installer manifests. The CR will have the `edge` on the name instead of `worker`:
```
$ oc get machinesets -n openshift-machine-api 
NAME                                         DESIRED   CURRENT   READY   AVAILABLE   AGE
ipi-local-zones-tzzwt-edge-us-west-2-lax-1a  3         3         3       3           34m
ipi-local-zones-tzzwt-edge-us-west-2-lax-1b  2         2         2       2           35m
ipi-local-zones-tzzwt-worker-us-west-2a      1         1         1       1           35m
ipi-local-zones-tzzwt-worker-us-west-2b      1         1         1       1           35m
ipi-local-zones-tzzwt-worker-us-west-2c      1         1         1       1           35m
```